pandas/core/reshape/reshape.py

```diff
@@ -711,7 +714,7 @@ def _convert_level_number(level_num: int, columns: Index):
             roll_columns = roll_columns.swaplevel(lev1, lev2)
         this.columns = mi_cols = roll_columns
 
-    if not mi_cols._is_lexsorted():
+    if not mi_cols._is_lexsorted() and sort:
```

Comment:
I'm wondering if this is behaving as intended. In the example below, I would think that the rows would be swapped (the stacked 2nd level of the index being 0 1 instead of 1 0)

```
levels = ((0, 1), (1, 0))
stack_lev = 1
columns = MultiIndex(levels=levels, codes=[[0, 0, 1, 1], [0, 1, 0, 1]])
df = DataFrame(columns=columns, data=[range(4)])
df_stacked = df.stack(stack_lev, sort=True)
print(df_stacked)
#      0  1
# 0 1  0  2
#   0  1  3
```

`mi_cols._is_lexsorted()` is checking if the _codes_ are lexsorted (they are `[[0, 0, 1, 1], [0, 1, 0, 1]]` here) but not if the values are sorted (they are `[[0, 1], [1, 0]]` here).

Should sort be sorting the level values?

Reply:
Nice find. Yeah I agree and I think the level values should be sorted here

Reply:
Thanks, I plan on putting up a PR for this.

Reply:
Thanks!